pkg/tfbridge/schema.go

```diff
@@ -355,13 +355,24 @@ func (ctx *conversionContext) makeTerraformInput(
 	tfs shim.Schema,
 	ps *SchemaInfo,
 ) (interface{}, error) {
-	// For TypeList or TypeSet with MaxItems==1, we will have projected as a scalar
-	// nested value, and need to wrap it into a single-element array before passing to
-	// Terraform.
-	if IsMaxItemsOne(tfs, ps) {
+	// For TypeList or Typeset with MaxItems==1, we will have projected as a scalar.
+	// If the MaxItemsOne is removed, we need still to correctly handle the scalar value
+	// and wrap it in an array.
+	// It is never correct to pass a scalar value to Terraform, where the schema indicates
+	// that a collection is expected.
+	if tfs != nil && (tfs.Type() == shim.TypeList || tfs.Type() == shim.TypeSet) {
 		wrap := func(val resource.PropertyValue) resource.PropertyValue {
 			if val.IsNull() {
-				return resource.NewArrayProperty([]resource.PropertyValue{})
+				// Keep the old behavior of presenting null maxItemsOne values as empty arrays.
+				if IsMaxItemsOne(tfs, ps) {
+					return resource.NewArrayProperty([]resource.PropertyValue{})
+				}
+
+				return val
+			}
+			// Do not attempt to wrap unknowns.
```

Comment:
Does this code ever deal with secrets? If not can we put an assert that pv is not secret?